### PR TITLE
PP-105: signature verification on createUserSmartWallet call

### PR DESCRIPTION
**Pull request by mortelli**
This PR is a `rif-relay-contracts` port of @patogallaiovlabs's original PR from the `rif-relay` repo (see [here](https://github.com/rsksmart/rif-relay/pull/178)).

A PR with tests is available here.

### CODE REVIEW

**Review by antomor — APPROVED** [review]
LGTM

**Review by antomor — APPROVED** [review]
LGTM

**Comment by antomor** [comment]
@franciscotobar I've rebased this PR, could you please review it when you can?

File: contracts/factory/CustomSmartWalletFactory.sol [hunk 1]
```diff
@@ -118,20 +118,17 @@ contract CustomSmartWalletFactory is ICustomSmartWalletFactory {
...
+            logic,
+            index,
+            initParams
+        ));
         (sig);
-        require(
-            RSKAddrValidator.safeEquals(keccak256(packed).recover(sig), owner),
-            "Invalid signature"
-        );
+        bytes32 message = keccak256(abi.encodePacked("\x19Ethereum Signed Message:\n32", _hash));
```
Comment (franciscotobar) [inline contracts/factory/CustomSmartWalletFactory.sol:130]: This change will affect the smartWalletFactoryTest(Line48) and CustomSmartWalletFactoryTest(Line50).

Comment (antomor) [reply contracts/factory/CustomSmartWalletFactory.sol:130]: @franciscotobar  Yes, indeed, thank you for pointing it out.
I've checked the failing errors and applied the required changes.

**Review by jurajpiar — APPROVED** [review]

**Review by ironFe93 — APPROVED** [review]

**Pull request merged**